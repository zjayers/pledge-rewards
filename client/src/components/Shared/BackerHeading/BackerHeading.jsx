import React from 'react';
import BackerHeading from './BackerHeading.style';

export default (props) => {
  const { activated, uppercase, children } = props;
  return (
    <BackerHeading activated={activated} uppercase={uppercase}>
      {children}
    </BackerHeading>
  );
};
