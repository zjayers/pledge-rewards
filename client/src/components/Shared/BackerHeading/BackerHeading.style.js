import styled from 'styled-components';

export default styled.p`
  margin-top: 20px;
  color: ${(props) => props.theme.colors.flavorText};
  text-transform: ${(props) => (props.uppercase ? 'uppercase' : 'lowercase')};
  margin-bottom: ${(props) => (props.activated ? '20px' : '0')};
`;
