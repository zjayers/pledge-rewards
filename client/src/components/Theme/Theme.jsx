import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
