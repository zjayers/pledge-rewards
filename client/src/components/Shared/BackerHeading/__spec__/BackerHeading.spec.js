import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import BackerHeading from '../BackerHeading';

import Theme from '../../../Theme/Theme';

describe('BackerHeading Component', () => {
  let component;
  let componentLowercase;
  beforeEach(() => {
    component = mount(
      <Theme>
        <BackerHeading activated uppercase />
      </Theme>
    );

    componentLowercase = mount(
      <Theme>
        <BackerHeading activated={false} uppercase={false} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
    componentLowercase.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot for uppercase & activated heading', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match the test snapshot for lowercase & de-activated heading', () => {
    expect(toJson(componentLowercase)).toMatchSnapshot();
  });
});
