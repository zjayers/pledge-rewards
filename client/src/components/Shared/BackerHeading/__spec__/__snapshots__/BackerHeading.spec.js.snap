// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackerHeading Component should match the test snapshot for lowercase & de-activated heading 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      activated={false}
      uppercase={false}
    >
      <BackerHeadingstyle___default
        activated={false}
        uppercase={false}
      >
        <p
          className="BackerHeadingstyle___default-t7usxk-0 lhsgcG"
        />
      </BackerHeadingstyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;

exports[`BackerHeading Component should match the test snapshot for uppercase & activated heading 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      activated={true}
      uppercase={true}
    >
      <BackerHeadingstyle___default
        activated={true}
        uppercase={true}
      >
        <p
          className="BackerHeadingstyle___default-t7usxk-0 kYjYVh"
        />
      </BackerHeadingstyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;
